# origin: Horms <horms@verge.net.au>
# backport cset: 1.1982.55.14 (2.6) key=4208e1fcfccuD-eH2OGM5mBhihmQ3A
# backport URL: http://linux.bkbits.net:8080/linux-2.6/cset@4208e1fcfccuD-eH2OGM5mBhihmQ3A
# inclusion: backport from 2.6, submitted upstream
# descrition: Fix ATM copy-to-user usage.
# revision date: Wed, 23 Mar 2005 12:47:43 +0900
#
# Backport of  ATM copy-to-user signedness fix from 2.6
#
# Signed-off-by: Simon Horman <horms@verge.net.au>

===== net/atm/addr.h 1.2 vs edited =====
--- 1.2/net/atm/addr.h	2002-02-05 16:39:14 +09:00
+++ edited/net/atm/addr.h	2005-03-23 13:40:46 +09:00
@@ -13,6 +13,6 @@
 void atm_reset_addr(struct atm_dev *dev);
 int atm_add_addr(struct atm_dev *dev,struct sockaddr_atmsvc *addr);
 int atm_del_addr(struct atm_dev *dev,struct sockaddr_atmsvc *addr);
-int atm_get_addr(struct atm_dev *dev,struct sockaddr_atmsvc *u_buf,int size);
+int atm_get_addr(struct atm_dev *dev,struct sockaddr_atmsvc *u_buf,size_t size);
 
 #endif
===== net/atm/addr.c 1.4 vs edited =====
--- 1.4/net/atm/addr.c	2003-09-04 12:31:04 +09:00
+++ edited/net/atm/addr.c	2005-03-23 13:41:03 +09:00
@@ -114,7 +114,7 @@
 }
 
 
-int atm_get_addr(struct atm_dev *dev,struct sockaddr_atmsvc *u_buf,int size)
+int atm_get_addr(struct atm_dev *dev,struct sockaddr_atmsvc *u_buf,size_t size)
 {
 	unsigned long flags;
 	struct atm_dev_addr *walk;
